.github/workflows/publish-qa.yml

```diff
@@ -30,28 +35,28 @@ jobs:
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v1
 
-      - name: Login to Docker Hub
+      - name: Login to the Docker registry
         uses: docker/login-action@v1
         with:
-          username: ${{ secrets.DOCKER_USERNAME }}
-          password: ${{ secrets.DOCKER_PASSWORD }}
+          registry: ${{ env.REGISTRY_HOST }}
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Get short SHA
-        id: get_sha
-        run: echo "::set-output name=short_sha::$(git rev-parse --short HEAD)"
+      - name: Generate tags for web-patron image
+        id: web-patron-tags
+        uses: docker/metadata-action@v3
+        with:
+          images: ${{ env.REGISTRY_HOST }}/${{ github.repository_owner }}/web-patron
+          tags: |
+            type=semver,pattern={{major}}.{{minor}},priority=10
+            type=semver,pattern={{version}},priority=20
+            type=ref,event=branch,priority=30
```

Comment:
I think these tag names rely on the push action, rather then the PR closed action. 
https://github.com/docker/metadata-action#tags-input

Reply:
Yes, you're right. I'll make this run on push instead.